### PR TITLE
aes: Implement aes-gcm128/256 with crypto extension

### PR DESCRIPTION
**Pull request by yuhaoth**

### CODE REVIEW

File: Makefile.unx [hunk 1]
```diff
@@ -44,10 +44,16 @@ endif
 default: lib
 include $(foreach unit,$(units), $(unit)/Makefile.am)
 
+ifeq ($(arch),aarch64)
+include aes/aarch64/Makefile.am
```
Comment (gbtucker) [inline Makefile.unx:48]: This is getting messy.  Why not just add `units += aes` to this if block and the last change with vpath will be unnecessary?

**Review by gbtucker — COMMENTED** [review]
Looks ok.  Just one suggestion.

File: Makefile.unx (hunk 1 above)
Comment (yuhaoth) [reply Makefile.unx:48]: XTS and CBC are not implemented. `units += aes ` will build XTS and CBC tests , that will break the build .

I will change it to `units += aes` after XTS and CBC tests done.

**Comment by gbtucker** [comment]
Integrated

**Pull request closed without merging**